### PR TITLE
DEV-1633: rework gem

### PR DESCRIPTION
**Pull request by maneyko**
Rework gem request and display.

### CODE REVIEW

**Comment by zbelzer** [comment]
What's the overarching goal here?

File: lib/slackiq.rb [hunk 1]
```diff
@@ -1,156 +1,192 @@
...
-      extra_fields = options.except(:webhook_name, :title, :description, :status)
+  # @param options [Hash]
+  def self.notify(options)
+    new(options).execute
+  end
 
-      fields = []
+  # @param options [Hash]
+  def self.configure(webhook_urls={})
+    @@webhook_urls = webhook_urls
```
Comment (zbelzer) [inline lib/slackiq.rb]: Rarely does one actually need "class variables". One `@` in a `self` context would signify it's a class instance variable and that is generally sufficient.

**Comment by maneyko** [comment]
I have verified the changes work in my most recent commit. Please re-review. Thanks.

File: HornsAndHooves-slackiq.gemspec [hunk 2]
```diff
@@ -1,8 +1,8 @@
 # -*- encoding: utf-8 -*-
 
-lib = File.expand_path("../lib", __FILE__)
+lib = File.expand_path("lib", __dir__)
 $LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
-require "slackiq/version"
```
Comment (zbelzer) [inline HornsAndHooves-slackiq.gemspec:5]: Typically gems store the version in their own file under <gemname>/version.rb. They usually only contain a version constant and that is fine.

**Pull request merged**